src/main/java/com/cloudogu/sslcontext/Certificate.java

```diff
@@ -74,6 +74,7 @@ private String createFingerprint(byte[] certificate) {
   }
 
   public void approve() {
+    timestamp = Instant.now();
```

Comment:
Why does the timestamp change if the status could not be changed? The timestamp should only updated, if the status was successfully updated.